alarms/alarms.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package alarms
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/absmach/supermq/pkg/authn"
+)
+
+const (
+	SeverityMax uint8 = 100
+)
+
+var ErrInvalidSeverity = errors.New("invalid severity. Must be between 0 and 100")
+
+type Metadata map[string]interface{}
+
+// Alarm represents an alarm instance.
+type Alarm struct {
+	ID          string    `json:"id"`
+	RuleID      string    `json:"rule_id"`
+	Status      Status    `json:"status"`
+	Measurement string    `json:"measurement"`
+	Value       string    `json:"value"`
+	Unit        string    `json:"unit"`
+	Cause       string    `json:"cause"`
+	Severity    uint8     `json:"severity"`
+	DomainID    string    `json:"domain_id"`
+	AssigneeID  string    `json:"assignee_id"`
+	CreatedAt   time.Time `json:"created_at"`
+	UpdatedAt   time.Time `json:"updated_at"`
```

Comment:
Can we also include a threshold value for the alarm here?

Reply:
Okay